packages/extension-ui/src/Popup/Derive/SelectParent.tsx

```diff
@@ -131,17 +147,20 @@ export default function SelectParent ({ className, isLocked, onDerivationConfirm
             <>
               <DerivationPath
                 defaultPath={defaultPath}
-                isError={!suriPath}
-                onChange={setSuriPath}
+                isError={!suriPath || !isValidPath}
+                onChange={_onSuriPathChange}
                 parentAddress={parentAddress}
                 parentPassword={parentPassword}
               />
-              {!suriPath && (
+              {(!suriPath || !isValidPath) && (
                 <Warning
                   isBelowInput
                   isDanger
                 >
-                  {t('Incorrect derivation path')}
+                  {!suriPath
+                    ? t('Incorrect derivation path')
+                    : t('`///password` not supported for derivation')
```

Comment:
Not 100% sure about this - there could be other errors? (I guess it is fine for a start)

Reply:
The other option, I thought about was to throw in the background process, when validating the suri. I would have to change a little the message api and add a `allowPassword`, but I found that odd to have a request + answer message for something we know beforehand, and that is only relevant for this screen.

Reply:
Yes, also had that brief thought atm - but it adds a loooong delay in actually making the request.

Reply:
The apps UI has this validation function -

```js

function deriveValidate (suri: string, pairType: KeypairType): string | null {
  if (suri.includes('///')) {
    return 'Password paths are not supported on keys derived from others';
  }

  try {
    const { path } = keyExtractPath(suri);

    // we don't allow soft for ed25519
    if (pairType === 'ed25519' && path.some(({ isSoft }): boolean => isSoft)) {
      return 'Soft derivation paths are not allowed on ed25519';
    }
  } catch (error) {
    console.error(error);

    return (error as Error).message;
  }

  return null;
}
```

Reply:
pairType there is actually incorrect, it should be `pairType !== sr25519`... (and then with a relevant change to the error)

Reply:
I remember checking it the other day. We also used to have the `keyExtractPath` in the UI. The reason I ditched it in the last PR is that it was a duplicate because also used in the background when creating an account.

I'll make the error handling a little more generic and add the check for sr25519.